src/pymovements/gaze/_utils/_parsing_begaze.py

```diff
@@ -0,0 +1,463 @@
+            normalized = line.rstrip('\n')
+            if '\t' in normalized:
+                header_cols = [c.strip() for c in normalized.split('\t')]
+            else:
+                # Space-separated headers cannot reliably represent multi-word column names
+                # like "L POR X [px]". We detect tracked eye from the raw string, but we
+                # will not attempt to parse samples from such headers.
+                header_cols = None
+            header_row_index = idx
+            # Determine tracked eye from presence of L/R POR columns
+            upper = normalized.upper()
+            if 'L POR X' in upper and 'R POR X' not in upper:
+                header_tracked_eye = 'L'
+            elif 'R POR X' in upper and 'L POR X' not in upper:
+                header_tracked_eye = 'R'
+            elif 'L POR X' in upper and 'R POR X' in upper:
+                header_tracked_eye = prefer_eye or 'L'
+            # build index map
+            header_idx = {name: i for i, name in enumerate(header_cols)} if header_cols else {}
+            break
+
+    if header_datetime is not None:
+        metadata['datetime'] = header_datetime
+    if header_sampling_rate is not None:
+        metadata['sampling_rate'] = header_sampling_rate
+    if header_tracked_eye is not None:
+        metadata['tracked_eye'] = header_tracked_eye
+
+    # metadata keys specified by the user should have a default value of None
+    metadata_keys = get_pattern_keys(compiled_metadata_patterns_prefixed, 'key')
+    for key in metadata_keys:
+        metadata[key] = None
+
+    # Determine which columns to use based on prefer_eye and availability
+    selected_eye = prefer_eye.upper() if prefer_eye else 'L'
+
+    def has_eye_columns(eye: str) -> bool:
+        return (
+            f'{eye} POR X [px]' in header_idx and
+            f'{eye} POR Y [px]' in header_idx and
+            f'{eye} Pupil Diameter [mm]' in header_idx
+        )
+
+    if not has_eye_columns(selected_eye):
+        # fall back to the other eye if available and inform the user once
+        other_eye = 'R' if selected_eye == 'L' else 'L'
+        if has_eye_columns(other_eye):
+            warnings.warn(
+                f"BeGaze parser: preferred eye '{selected_eye}' not found in columns; "
+                f"falling back to '{other_eye}'.",
+                RuntimeWarning,
+            )
+            selected_eye = other_eye
+
+    # Decide if we can parse samples: requires tabular header with eye columns
+    parse_samples = False
+    if header_row_index is not None and header_cols is not None:
+        parse_samples = has_eye_columns('L') or has_eye_columns('R')
+
+    # iterate over data lines following the header row
+    assert header_row_index is not None
+    for line in lines[header_row_index + 1:]:
+        # Apply message-driven additional columns first
+        for pattern_dict in compiled_patterns:
```

Comment:
in this case a dedicated preprocessing method could also help with this in the future and offload some logic.

What we would need for this is transforming `messages` to `samples` and `events` (related to the sidenote in #1290).